### DISCUSSION TITLE
Add Agentic Reinforcement Learning (RL) Support to strands-agents

### DISCUSSION THREAD

**Discussion by sharabhshukla** [strands-agents/sdk-python#597] [Ideas]
I'd like to propose adding agentic reinforcement learning (RL) support to the strands-agents framework. This would allow agents to adapt over time via feedback, optimizing their decision-making and task execution policies based on reward signals — not just static prompting or rule-based control.

By integrating reinforcement learning mechanisms (e.g. PPO, DPO), strands-agents could support training loops where agents learn from interaction, user feedback, or downstream outcomes — especially useful in production AWS environments.

The strands-agents project is a powerful foundation for building autonomous LLM agents that interact with tools, memories, and APIs. However, the current system is prompt-centric and static — agents are programmed via tool definitions and planning logic, but don't yet have the ability to:

Improve themselves via reward-based feedback

Learn from failures or corrections over time

Optimize behaviors for specific long-term objectives

Adding RL-based learning capabilities would unlock a new class of adaptive, continuously improving agents, aligned with real-world goals.

A plug-and-play reinforcement learning interface, compatible with existing AgentController-based agents:

RLTrainer(agent, environment, reward_fn).train()

'd love to hear community thoughts on this 

Does agentic RL align with the vision for strands-agents?

Would a module like strands-agents-rl make sense as a first step?

Is there existing internal work on integrating RL-based learning into AWS agent frameworks?

Would the community be interested in collaborating on a prototype?

**Comment by JackYPCOnline** [comment]
Hi there,
Thank you for bringing this up. After you collect enough information and believe it's a strong use case, you're very 
welcome to open a feature request. From there, our team will evaluate it.

**Reply by sharabhshukla** [reply]
Some folks, at Microsoft have created, https://github.com/microsoft/agent-lightning. It would be nice, if AWS in the strands framework makes it easier to fine tune an LLM for agentic tasks. This also goes well with the idea that AWS being a massive compute provider.

**Comment by theagenticguy** [comment]
This may be related, https://github.com/strands-agents/sdk-python/issues/609
